src/test/scala/me/jeffmay/neo4j/client/ws/WSNeo4jClientBasicSpecs.scala

```diff
@@ -0,0 +1,94 @@
+package me.jeffmay.neo4j.client.ws
+
+import me.jeffmay.neo4j.client._
+import me.jeffmay.util.UniquePerClassNamespace
+import me.jeffmay.util.akka.TestGlobalAkka
+import org.scalatest._
+import play.api.libs.json.Json
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class WSNeo4jClientBasicSpecs extends fixture.AsyncWordSpec
+  with Safety
+  with UniquePerClassNamespace
+  with CleanupNamespace {
+
+  override implicit def executionContext: ExecutionContext = ExecutionContext.global
+
+  class FixtureParam extends UniqueNamespace {
+    val client: Neo4jClient = new WSNeo4jClient(
+      TestNeo4jWSClient.TestWS,
+      Neo4jClientConfig.fromConfig(TestGlobal.config),
+      TestGlobalAkka.scheduler,
+      executionContext
+    )
+  }
+
+  override def withAsyncFixture(test: OneArgAsyncTest): Future[Outcome] = {
+    val fixture = new FixtureParam
+    val futureOutcome = test(fixture)
+    val x = futureOutcome.andThen {
+      case _ => cleanupNamespace(fixture.namespace, fixture.client)
+    }
+    x
+  }
+
+  "WSNeo4jClient" when {
+
+    "providing authorization" should {
+
+      "not require a password change when auto-initialize is turned on" in { fixture =>
+        import fixture._
+        for {
+          passwordChangeRequired <- client.passwordChangeRequired()
+        } yield {
+          assertResult(false, "password change should not be required after initializing") {
+            passwordChangeRequired
+          }
+        }
+      }
+    }
+
+    "calling openAndCloseTxn" should {
+
+      "start with an empty unique namespace" in { fixture =>
+        import fixture._
+        val getAllNodes = CypherStatement(
+          "MATCH (n { ns: {props}.ns }) RETURN n",
+          CypherParameters(Json.obj("ns" -> namespace))
+        )
+        for {
+          rsp <- client.openAndCommitTxn(Seq(getAllNodes))
+        } yield {
+          assertResult(true, "should not encounter errors") {
+            rsp.errors.isEmpty
+          }
+          assertResult(Seq(), "database should be empty") {
+            // TODO: Results by index is ok, but not as ideal as a map
+            rsp.results.head.data
+          }
+        }
+      }
+
+      "add a single node to the graph" in { fixture =>
+        import fixture._
+        val createNode = CypherStatement(
+          "CREATE(n { ns: {props}.ns })",
+          CypherParameters(Json.obj("ns" -> namespace))
+        )
+        for {
+          rsp <- client.withStatsIncludedByDefault(true).openAndCommitTxn(Seq(createNode))
+        } yield {
+          assertResult(true, "should not encounter errors") {
+            rsp.errors.isEmpty
+          }
+          // TODO: make this work by parsing the stats
+//          assertResult(1, "should have created 1 node") {
+//            rsp.results.head.stats.get.nodesCreated
+//          }
```

Comment:
This would be good to have before closing this PR. I already have the code for it, I just need to make use of the given boolean config.